types/shared.d.ts

```diff
@@ -8,7 +8,9 @@ export interface PageData {
   titleTemplate?: string | boolean
   description: string
   headers: Header[]
-  frontmatter: Record<string, any>
+  frontmatter: Record<string, any> & {
+    outline?: number | [number, number] | 'deep'
```

Comment:
allow it to be disabled too?

Reply:
Also, can it be configured through themeConfig too so that users don't need to put that on every page?

Reply:
added `false` to disable the outline